src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts

```diff
@@ -1884,8 +1884,8 @@ class TaskService extends Disposable implements ITaskService {
 			}
 		};
 		return this.quickInputService.pick(_createEntries().then((entries) => {
-			if (entries.length === 0 && defaultEntry) {
-				entries.push(defaultEntry);
+			if (defaultEntry) {
```

Comment:
The `showQuickPick` is called from many different places and no inserting the default entry always at the front might break other places. Have you considered this?

Reply:
IMo this needs to be addressed here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts#L2336 and the tasks still need to be listed with the default task being at top.

Reply:
I checked where showQuickPick is called from and didn't see anywhere that this would be a problem. I will double check.

Reply:
I checked again. The only place that a default entry is used is in the configure build and test tasks. Do you still prefer that this be addressed at https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts#L2336  ?

Reply:
Actually I would. Main reason is that it will be easier to understand later on. The idea of the default entry for `showQuickPick` was to show it when no other elements are provided.